plugins/by-name/kulala/default.nix

```diff
@@ -0,0 +1,251 @@
+            ];
+            pathresolver = "require('kulala.parser.jsonpath').parse";
+          };
+          "application/xml" = {
+            ft = "xml";
+            formatter = [
+              "xmllint"
+              "--format"
+              "-"
+            ];
+            pathresolver = [
+              "xmllint"
+              "--xpath"
+              "{{path}}"
+              "-"
+            ];
+          };
+          "text/html" = {
+            ft = "html";
+            formatter = [
+              "xmllint"
+              "--format"
+              "--html"
+              "-"
+            ];
+            pathresolver = [ ];
+          };
+        }
+        ''
+          Defines formatters and path resolvers for specific content types.
+        '';
+
+    show_icons =
+      defaultNullOpts.mkEnum
+        [
+          "on_request"
+          "above_request"
+          "below_request"
+          null
+        ]
+        "on_request"
+        ''
+          Determines where icons are displayed relative to requests. Options are "on_request", "above_request", "below_request", or null to disable.
+        '';
+
+    icons =
+      defaultNullOpts.mkAttrsOf lib.types.anything
+        {
+          inlay = {
+            loading = "⏳";
+            done = "✅";
+            error = "❌";
+          };
+          lualine = "🐼";
+        }
+        ''
+          Sets default icons for loading states and lualine.
+        '';
+
+    scratchpad_default_contents =
+      defaultNullOpts.mkListOf lib.types.anything
+        [
+          "@MY_TOKEN_NAME=my_token_value"
+          ""
+          "# @name scratchpad"
+          "POST https://httpbin.org/post HTTP/1.1"
+          "accept: application/json"
+          "content-type: application/json"
+        ]
+        ''
+          Default contents for the scratchpad feature.
+        '';
+
+    winbar = defaultNullOpts.mkBool false ''
+      Enables or disables the winbar for displaying request status.
+    '';
+
+    default_winbar_panes =
+      defaultNullOpts.mkListOf lib.types.str
+        [
+          "body"
+          "headers"
+          "headers_body"
+        ]
+        ''
+          Specifies which panes are displayed by default in the winbar.
+        '';
+
+    vscode_rest_client_environmentvars = defaultNullOpts.mkBool false ''
+      Enables reading environment variables from VSCode’s REST client.
+    '';
+
+    disable_script_print_output = defaultNullOpts.mkBool false ''
+      Disables immediate printing of script outputs; outputs will still be written to disk.
+    '';
+
+    environment_scope =
+      defaultNullOpts.mkEnum
+        [
+          "b"
+          "g"
+        ]
+        "b"
+        ''
+          Sets the scope for environment and request variables. Options are "b" (buffer) or "g" (global).
+        '';
+
+    certificates = defaultNullOpts.mkAttrsOf lib.types.anything { } ''
+      Configures SSL/TLS certificates.
+    '';
+  };
+
+  settingsExample = {
```

Comment:
Ideally, the example should be more concise.
It will be embedded in the description of the `settings` option.

Reply:
I removed a section for the contenttypes, are there any other recommendations on making it more concise?

Reply:
I was more thinking of ~5-6 options only.
You can look for
`language:lua require('kulala').setup` on GitHub to look at people's config.

Reply:
Got it, I reduced the settingsExample with fewer options similar to other example configs. Thanks @GaetanLepage !